### PR TITLE
Add debug logs for server checks

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
This PR adds an ability to get server checks logs with debug log level.

### What issues does this PR fix or reference?
It's a small step forward into the direction of https://github.com/eclipse/che/issues/16451

### How to test this PR?
I did not manage to test it, it's why it's a draft.
These logs are supposed to appear after we configure: `CHE_LOGGER_CONFIG=org.eclipse.che.api.workspace.server.hc=DEBUG`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dkwakkel** [comment]
I had to debug a similar issue and found out that it really helps to also have the url in the debug output. So maybe you can also apply this fix on top of yours:
``
diff --git a/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java b/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java
index 3d36e13530..5157c87585 100644
--- a/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java
+++ b/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java
@@ -103,8 +103,9 @@ public class HttpConnectionServerChecker extends ServerChecker {
           response = "failed to ready response: " + e.getMessage();
         }
         LOG.debug(
-            "Server check for '{}' request failed with code {}. Response: {}",
+            "Server check for '{}:{}' request failed with code {}. Response: {}",
             serverRef,
+            url,
             responseCode,
             response);
       }
@@ -112,8 +113,9 @@ public class HttpConnectionServerChecker extends ServerChecker {
       return success;
     } catch (IOException e) {
       LOG.debug(
-          "Failed to establish http connection to check server '{}'. Cause: {}",
+          "Failed to establish http connection to check server '{}:{}'. Cause: {}",
           serverRef,
+          url,
           e.getMessage());
       return false;
     }
``

**Comment by sleshchenko** [comment]
@dkwakkel I don't think I'll be able to test these changes soon. Would you like to apply your changes on top of my, test and make this PR ready to review? ) Feel free to create a new branch in your fork and create a new PR.

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java [hunk 1]
```diff
@@ -71,12 +84,43 @@ public boolean isAvailable() {
...
+          InputStream in = conn.getErrorStream();
+          if (in == null) {
+            in = conn.getInputStream();
+          }
+
+          try (Reader reader = new InputStreamReader(in)) {
+            response = CharStreams.toString(reader);
+          }
+        } catch (Exception e) {
+          response = "failed to ready response: " + e.getMessage();
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java:101]: do we need to re-throw `Exception` here ?

**Review by skabashnyuk — COMMENTED** [review]
in general, looks good

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java (hunk 1 above)
Comment (sleshchenko) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/HttpConnectionServerChecker.java:101]: Nope. It's server check where we just read the response and retry later when it fails.

**Comment by sleshchenko** [comment]
Taking into account https://github.com/eclipse/che/issues/18360#issuecomment-728682807, it would be cool to even improve that approach and either make these logs by default info but not debug level **OR** detect the last try and log as many details as we have on the last try failure.
It can be done separately from the current PR.

**Review by skabashnyuk — APPROVED** [review]

**Review by metlos — APPROVED** [review]

**Comment by skabashnyuk** [comment]
works for me on image `quay.io/skabashn/che-server:debugServerChecks`

Examples

```

│ 2020-11-24 13:51:17,987[ServersChecker]   [DEBUG] [.h.HttpConnectionServerChecker 103]  - Server check for 'theia:https://routei1lqj6o6-opentlc-mgr-che.apps.cluster-devtools-e990.devtools-e990.example.ope │
│ ntlc.com' request failed with code 500. Response: dial tcp 127.0.0.1:3100: connect: connection refused


```
or
```
 2020-11-24 13:56:09,836[ServersChecker]   [DEBUG] [.h.HttpConnectionServerChecker 103]  - Server check for 'theia:https://routek4wz2g71-opentlc-mgr-che.apps.cluster-devtools-e990.devtools-e990.example.ope │
│ ntlc.com' request failed with code 503. Response: <html>                                                                                                                                                     │
│   <head>                                                                                                                                                                                                     │
│     <meta name="viewport" content="width=device-width, initial-scale=1">                                                                                                                                     │
│                                                                                                                                                                                                              │
│   <style type="text/css">                                                                                                                                                                                    │
│   /*!                                                                                                                                                                                                        │
│    * Bootstrap v3.3.5 (http://getbootstrap.com)                                                                                                                                                              │
│    * Copyright 2011-2015 Twitter, Inc.                                                                                                                                                                       │
│    * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)                                                                                                                              │
│    */                                                                                                                                                                                                        │
│   /*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */        
```

**Pull request merged**